docs/reference-guides/interactivity-api/core-concepts/the-reactive-and-declarative-mindset.md

```diff
@@ -0,0 +1,305 @@
+    } else {
+      statusParagraph.textContent = 'this is inactive';
+      statusParagraph.classList.remove('active');
+      statusParagraph.classList.add('inactive');
+      activateBtn.textContent = 'activate';
+    }
+  });
+</script>
+```
+
+As you can see, for each condition, we have to use JavaScript to modify everything in the DOM that has changed taking into account the previous state.
+
+### Declarative Approach
+
+The declarative approach simplifies the process by focusing on _what_ should happen. The UI updates automatically in response to changes in state. Here is a similar example using the Interactivity API's declarative approach:
+
+```html
+<!-- Render.php file -->
+<button
+  data-wp-on--click="actions.toggleVisibility"
+  data-wp-bind--aria-expanded="state.isVisible"
+  data-wp-text="state.visibilityText"
+  aria-controls="status-paragraph"
+>
+  show
+</button>
+<button
+  data-wp-on--click="actions.toggleActivation"
+  data-wp-bind--disabled="!state.isVisible"
+  data-wp-text="state.activationText"
+>
+  activate
+</button>
+<p
+  id="status-paragraph"
+  data-wp-bind--hidden="!state.isVisible"
+  data-wp-class--active="state.isActive"
+  data-wp-class--inactive="!state.isActive"
+  data-wp-text="state.paragraphText"
+>
+  this is inactive
+</p>
+```
+
+```js
+// view.js file
+import { store } from '@wordpress/interactivity';
+
+const { state } = store('myPlugin', {
+  state: {
+    isVisible: false,
+    isActive: false,
+    get visibilityText() {
+      return state.isVisible ? 'hide' : 'show';
+    },
+    get activationText() {
+      return state.isActive ? 'deactivate' : 'activate';
+    },
+    get paragraphText() {
+      return state.isActive ? 'this is active' : 'this is inactive';
+    },
+  },
+  actions: {
+    toggleVisibility() {
+      state.isVisible = !state.isVisible;
+      if (!state.isVisible) state.isActive = false;
+    },
+    toggleActivation() {
+      state.isActive = !state.isActive;
+    },
+  },
+});
+```
+
+In this declarative example, the UI automatically updates based on the current state. All we have to do as developers is to declare the necessary state, any derived state, the actions that modify the state, and which parts of the DOM depend on which parts of the state. The framework takes care of making all the necessary updates to the DOM so that it is always in sync with the current state. The logic remains simple and maintainable regardless of the number of elements controlled by the framework.
+
+### Can you spot the bug?
+
+In the imperative example, we intentionally introduced a bug. Can you find it? It's not easy!
+
+<details>
+<summary>Show me the answer!</summary>
+
+In the case that the Show button is pressed first, then the Activate button, and finally the Hide button, we forgot to add the `inactive` class using `statusParagraph.classList.add('inactive');`. Therefore, the next time the user presses Show, the paragraph will not appear in red.
```

Comment:
Nothing critical but to avoid the use of "we" in the docs, maybe we can say here:

> In the case that the Show button is pressed first, then the Activate button, and finally the Hide button, there's a missing step in the previous code for adding the `inactive` class using `statusParagraph.classList.add('inactive');`. Therefore, the next time the user presses Show, the paragraph will not appear in red.